File: WeatherForecast/WeatherForecast/ViewControllers/AddCityVC.m
//
//  AddCityVC.m
//  WeatherForecast
//
//  Created by Hakkı Yiğit Yener on 23.05.2020.
//  Copyright © 2020 N11.com. All rights reserved.
//

#import "AddCityVC.h"
#import "City.h"
#import "CityManager.h"
#import "UITextField+Extensions.h"
#import "UIViewController+Extensions.h"

@interface AddCityVC ()<UIPickerViewDelegate,UIPickerViewDataSource>
@property (weak, nonatomic) IBOutlet UITextField *cityNameTF;
@property (weak, nonatomic) IBOutlet UITextField *reportDayCountTF;
@property (strong, nonatomic) UIPickerView *dayPicker;

@end

@implementation AddCityVC

- (void)viewDidLoad {
    [super viewDidLoad];
    // Do any additional setup after loading the view.
    [self setupUI];
}

-(void)setupUI {
    self.dayPicker = [self.reportDayCountTF addPickerView:self
                                          dataSource:self];
    [self.cityNameTF addToolbarTarget:self
                               action:@selector(pickerDoneButtonTapped:)];
    [self.reportDayCountTF addToolbarTarget:self
                                     action:@selector(pickerDoneButtonTapped:)];
}

-(void)pickerDoneButtonTapped:(id)picker{
    [self.view endEditing:YES];
}

- (IBAction)closeButtonTouched:(id)sender {
    [self dismissViewControllerAnimated:YES
                             completion:nil];
}
- (IBAction)saveButtonTouched:(id)sender {
    [self.view endEditing:YES];
    
    if (self.cityNameTF.text.length == 0) {
        [self showAlertWithTitle:@"Uyarı"
                         message:@"Şehir adı boş olamaz."];
        return;
    }
    
    City* newCity = [[City alloc] init];
    newCity.name = self.cityNameTF.text;
    newCity.reportDayCount = [self.reportDayCountTF.text intValue];
    [[CityManager shared] addCity:newCity];
    [self dismissViewControllerAnimated:YES
                             completion:nil];
}

- (NSInteger)numberOfComponentsInPickerView:(nonnull UIPickerView *)pickerView {
    return 1;
}

- (NSInteger)pickerView:(nonnull UIPickerView *)pickerView numberOfRowsInComponent:(NSInteger)component {
    return 16;
}

-(NSString *)pickerView:(UIPickerView *)pickerView titleForRow:(NSInteger)row forComponent:(NSInteger)component {
    return [NSString stringWithFormat:@"%li",row + 1];
}

- (void)pickerView:(UIPickerView *)pickerView didSelectRow:(NSInteger)row inComponent:(NSInteger)component {
    self.reportDayCountTF.text = [NSString stringWithFormat:@"%li",row + 1];
}

- (BOOL)textFieldShouldReturn:(UITextField *)textField
{
    if (self.cityNameTF == textField) {
        [self.reportDayCountTF becomeFirstResponder];
        return NO;
    }
    return YES;
}

@end

File: WeatherForecast/WeatherForecast/ViewControllers/AddCityVC.swift
//
//  AddCity.swift
//  WeatherForecast
//
//  Created by Hakkı Yiğit Yener on 27.05.2020.
//  Copyright © 2020 N11.com. All rights reserved.
//

import UIKit

class AddCityVC:UIViewController, UIPickerViewDelegate, UIPickerViewDataSource {
    
    @IBOutlet private weak var cityNameTF: UITextField!
    @IBOutlet private weak var reportDayCountTF: UITextField!
    private var dayPicker: UIPickerView?
    
    override func viewDidLoad() {
        super.viewDidLoad()
        // Do any additional setup after loading the view.
        setupUI()
    }

    func setupUI() {
        dayPicker = reportDayCountTF.addPickerView(
            self,
            dataSource: self)

        _ = cityNameTF.addToolbar(
            self,
            action: #selector(toolbarDoneButtonTapped(_:)))
        _ = reportDayCountTF.addToolbar(
            self,
            action: #selector(toolbarDoneButtonTapped))
    }
// MARK: - Action Methods
    @objc func toolbarDoneButtonTapped(_ picker: Any?) {
        view.endEditing(true)
    }

    @IBAction func closeButtonTouched(_ sender: Any) {
        dismiss(
            animated: true)
    }

    @IBAction func saveButtonTouched(_ sender: Any) {
        view.endEditing(true)

        if cityNameTF.text?.count == 0 {
            showAlert(
                withTitle: "Uyarı",
                message: "Şehir adı boş olamaz.")
            return
        }

        let newCity = City()
        newCity.name = cityNameTF.text
        newCity.reportDayCount = Int32.init(reportDayCountTF.text ?? "0")!
        _ = CityManager.shared.addCity(newCity)
        dismiss(
            animated: true)
    }
// MARK: - UIPickerView Methods
    func numberOfComponents(in pickerView: UIPickerView) -> Int {
        return 1
    }

    func pickerView(_ pickerView: UIPickerView, numberOfRowsInComponent component: Int) -> Int {
        return 16
    }

    func pickerView(_ pickerView: UIPickerView, titleForRow row: Int, forComponent component: Int) -> String? {
        return String(format: "%li", row + 1)
    }

    func pickerView(_ pickerView: UIPickerView, didSelectRow row: Int, inComponent component: Int) {
        reportDayCountTF.text = String(format: "%li", row + 1)
    }
// MARK: - UITextfield Methods
    func textFieldShouldReturn(_ textField: UITextField) -> Bool {
        if cityNameTF == textField {
            reportDayCountTF.becomeFirstResponder()
            return false
        }
        return true
    }
}
